e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js

```diff
@@ -1179,3 +1180,101 @@ describe("issue 49882", () => {
       .and("have.css", "background-color", "rgb(80, 158, 227)");
   });
 });
+
+describe("issue 49304", () => {
+  const questionDetails = {
+    query: {
+      "source-table": PRODUCTS_ID,
+    },
+  };
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("should be possible to switch between filter widgets and the expression editor for multi-argument operators (metabase#49304)", () => {
+    createQuestion(questionDetails, { visitQuestion: true });
+    openNotebook();
+
+    cy.log(
+      "add a filter using a filter widget and check that it is rendered in the expression editor",
+    );
+    getNotebookStep("data").button("Filter").click();
+    popover().findByText("Category").click();
+    selectFilterOperator("Contains");
+    popover().within(() => {
+      cy.findByPlaceholderText("Enter some text").type("gadget,widget");
```

Comment:
shouldn't we test parts of the filters and startsWith, endsWith?

at least steps from the issue show the problem with partial words

Reply:
I added unit tests for all other functions